resources/Microsoft.DotNet.Dsc/Microsoft.DotNet.Dsc.psm1

```diff
@@ -0,0 +1,547 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+$PSNativeCommandUseErrorActionPreference = $true
+Set-StrictMode -Version Latest
+
+#region Functions
+function Get-DotNetPath
+{
+    if ($IsWindows)
+    {
+        $dotNetPath = "$env:ProgramFiles\dotnet\dotnet.exe"
+        if (-not (Test-Path $dotNetPath))
+        {
+            $dotNetPath = "${env:ProgramFiles(x86)}\dotnet\dotnet.exe"
+            if (-not (Test-Path $dotNetPath))
+            {
+                throw "dotnet.exe not found in Program Files or Program Files (x86)"
+            }
+        }
+    }
+    elseif ($IsMacOS)
+    {
+        $dotNetPath = "/usr/local/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath))
+        {
+            $dotNetPath = "/usr/local/bin/dotnet"
+            if (-not (Test-Path $dotNetPath))
+            {
+                throw "dotnet not found in /usr/local/share/dotnet or /usr/local/bin"
+            }
+        }
+    }
+    elseif ($IsLinux)
+    {
+        $dotNetPath = "/usr/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath))
+        {
+            $dotNetPath = "/usr/bin/dotnet"
+            if (-not (Test-Path $dotNetPath))
+            {
+                throw "dotnet not found in /usr/share/dotnet or /usr/bin"
+            }
+        }
+    }
+    else
+    {
+        throw "Unsupported operating system"
+    }
+
+    Write-Verbose -Message "'dotnet' found at $dotNetPath"
+    return $dotNetPath
+}
+
+function Get-DotNetToolArguments
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $PackageId,
+        [Parameter(Mandatory = $false)]
+        [string] $Version,
+        [Parameter(Mandatory = $false)]
+        [bool]   $PreRelease,
+        [Parameter(Mandatory = $false)]
+        [string] $ToolPath,
+        [bool]   $Exist
+    )
+
+    $arguments = @($PackageId)
+
+    if (-not ($PSBoundParameters.ContainsKey("ToolPath")))
+    {
+        $arguments += "--global"
+    }
+
+    # mapping table of command line arguments
+    $mappingTable = @{
+        Version    = "--version {0}"
+        PreRelease = "--prerelease"
+        ToolPath   = "--tool-path {0}"
+    }
+    
+    $PSBoundParameters.GetEnumerator() | ForEach-Object {
+        if ($mappingTable.ContainsKey($_.Key) -and $_.Value -ne $false)
```

Comment:
Does `$.Value -ne $false` work for parameters that are not boolean parameters like $Version?

Reply:
Fixed with one more additional check